R/model_builder.R

```diff
@@ -102,15 +102,39 @@ build_kmeans.kv_ <- function(df,
   model_column <- avoid_conflict(grouped_column, "model")
   source_column <- avoid_conflict(grouped_column, "source.data")
 
+  df <- tidyr::drop_na_(df, c(subject_col, key_col, value_col))
+
   if(row_col %in% grouped_column){
     stop(paste0(row_col, " is a grouping column. ungroup() may be necessary before this operation."))
   }
 
   build_kmeans_each <- function(df){
     mat <- simple_cast(df, row_col, col_col, value_col, fun.aggregate = fun.aggregate, fill=fill)
-    kmeans_ret <- kmeans(mat, centers = centers, iter.max = 10, nstart = nstart, algorithm = algorithm, trace = trace)
+    kmeans_ret <- tryCatch({
+      kmeans(mat, centers = centers, iter.max = 10, nstart = nstart, algorithm = algorithm, trace = trace)},
+      error = function(e){
+        if(e$message == "cannot take a sample larger than the population when 'replace = FALSE'"){
+          # this matrix falls into here when centers = 3 and mat is
```

Comment:
I appreciate having example, but this example confuses me.
Is it an error because there are only group1 and group2 while centers is 3,
or is it because there are only 1 and 5 in the matrix cells while centers is 3,
or for some other reason?